File: kid/kid.py
# Python target module for MEOW.
# Copyright nergzd723
# Kills process by pid or by name
# GNU LGPL v3
import os
import sys
import signal
import psutil
ar = sys.argv[1:]
if not ar:
    print("kid MEOW utility\nKills process by pid(usage - kid (term or kill) pid)\n\
        Or by name(usage - kid -n nameof)")
    exit(1)
if '-n' in ar:
    for proc in psutil.process_iter():
        if proc.name() == ar[1]:
            proc.kill()
            print("Killed", proc.name(), 'with PID', proc.pid())
else:
    if 'term' in ar:
        ar = ar[1:]
        try:
            os.kill(ar[0], signal.SIGTERM)
            print("Killed! pid = {} signal = SIGTERM".format(ar[0]))
        except:
            print("No such process or access denied!")
    else:
        try:
            os.kill(ar[0], signal.SIGKILL)
            print("Killed! pid = {} signal = SIGKILL".format(ar[0]))
        except:
            print("No such process or access denied!")

File: kid/kid.cpp
// Meow part
// KID : Kills by ID
// Copyright 2019, Oleg Sazonov(@x0r3d) in collaboration with Mark Hargreaves(@nergzd723)
// 24.09.19: Initial write
#include <iostream>
#include <signal.h>
#include <unistd.h>
#include <cctype>
using namespace std;
int get_external_process_id( std::string name )
{
std::string output = "pidof ";
output += name; 
char line[ 15 ];
FILE *cmd = popen( output.c_str( ), "r");
fgets( line, 15, cmd );
pid_t pid = strtoul( line, NULL, 10 );
pclose(cmd);
return( pid );
};
int main(int argc, char *argv[]) {
    if(argc == 1) {
        cout << "KID: Kills by ID" << endl;
        cout << "Usage: kid name_of_process[pid]" << endl;
        return 0;
    };
    string kid = argv[1];
    pid_t pid;
    if(isdigit(atoi(kid.c_str())) && !kid.empty()) {
        pid = atoi(kid.c_str());
    } else if(!kid.empty()) {
        pid = get_external_process_id(kid);
    };
    int ret = kill(pid, SIGKILL);
    if(ret == 0) {
        cout << "It's don't killed!" << endl;
    } else {
        cout << "It's killed!" << endl;
    };
    return 0;
};